src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -9925,18 +9925,16 @@ static bool checkConstraints(CSharpCompilation compilation, ConversionsBase conv
                 return false;
             }
 
-            if (location == null)
-            {
-                location = node.Location;
-            }
-
             if (diagnosticInfo.Code == (int)ErrorCode.ERR_InvalidDelegateType)
             {
-                diagnostics.Add(new CSDiagnostic(new CSDiagnosticInfo(ErrorCode.ERR_InvalidDelegateType, possibleDelegateType), location));
+                diagnostics.Add(new CSDiagnostic(new CSDiagnosticInfo(ErrorCode.ERR_InvalidDelegateType, possibleDelegateType), getErrorLocation()));
                 return true;
             }
 
-            return Symbol.ReportUseSiteDiagnostic(diagnosticInfo, diagnostics, location);
+            return Symbol.ReportUseSiteDiagnostic(diagnosticInfo, diagnostics, getErrorLocation());
+
+            Location getErrorLocation()
+                => location ??= GetAnonymousFunctionLocation(node);
```

Comment:
>??= [](http://example.com/codeflow?start=28&length=3)

Why do we need the assignment part? Seems like `??` would be sufficient.

Reply:
Sure:-)